### PR TITLE
feat(Tearsheet): Slug to aiLabel

### LINKED ISSUE #6096: Tearsheet: `slug` --> `aiLabel` renaming
```[tasklist]
### Tasks
- [ ] Change slug prop to aiLabel
- [ ] Deprecate slug prop
- [ ] Add new duplicate classname selectors to the ailabel.scss styles
- [ ] Update classnames within AILabel components to use the new #{$prefix}--ailabel names
- [ ] Create a new sub-issue/tasklist item on https://github.com/carbon-design-system/ibm-products/issues for removing these classes as part of new major version
```

For duplicate class, this is an example:
```
- .#{$prefix}--slug .#{$prefix}--slug__button {

+ .#{$prefix}--slug .#{$prefix}--slug__button,
+ .#{$prefix}--ai-label .#{$prefix}--ai-label__button {
```

Image

### PR DESCRIPTION
**Pull request by sangeethababu9223**
Closes #6096

Rename the prop `slug` to `aiLabel`.

#### What did you change?

- Added Prop `aiLabel`
- Deprecated `slug` prop
- Updated the style with aiLabel classes as per the [ Core carbon standard](https://github.com/carbon-design-system/carbon/issues/17714)

#### How did you test and verify your work?
Storybook

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
   ],
   "words": [
     "aboutmodal",
+    "accountibility",
```
Comment (elycheea) [inline cspell.json:67]:
```suggestion
```

File: cspell.json [hunk 2]
```diff
@@ -155,6 +157,7 @@
     "rowindex",
     "ruleset",
     "sbdocs",
+    "scenerio",
```
Comment (elycheea) [inline cspell.json:160]:
```suggestion
```

File: packages/ibm-products/src/components/Tearsheet/Tearsheet.stories.jsx [hunk 3]
```diff
@@ -102,15 +102,29 @@ export default {
...
+      control: {
+        type: 'select',
+        labels: {
+          0: 'No AI Label',
+          1: 'with AI Label',
+        },
+        default: 0,
+      },
+      description:
+        'Optional prop that is intended for any scenerio where something is being generated by AI to reinforce AI transparency, accountibility, and explainability at the UI level.',
```
Comment (elycheea) [inline packages/ibm-products/src/components/Tearsheet/Tearsheet.stories.jsx:115]:
```suggestion
        'Optional prop that is intended for any scenario where something is being generated by AI to reinforce AI transparency, accountability, and explainability at the UI level.',
```

File: packages/ibm-products/src/components/Tearsheet/Tearsheet.stories.jsx [hunk 4]
```diff
@@ -102,15 +102,29 @@ export default {
...
       control: {
         type: 'select',
         labels: {
-          0: 'No AI slug',
+          0: 'No AI Slug',
           1: 'with AI Slug',
         },
         default: 0,
       },
+      description: 'deprecated Property replaced by "aiLabel"',
```
Comment (elycheea) [inline packages/ibm-products/src/components/Tearsheet/Tearsheet.stories.jsx:127]:
```suggestion
      description: 'Deprecated: Property replaced by "aiLabel"',
```

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx [hunk 5]
```diff
@@ -52,6 +52,12 @@ const maxDepth = 3;
 interface TearsheetShellProps extends PropsWithChildren {
   actions?: ButtonProps<'button'>[];
 
+  /**
+   * Optional prop that is intended for any scenerio where something is being generated by AI to reinforce AI transparency,
+   * accountibility, and explainability at the UI level.
```
Comment (elycheea) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx:57]:
```suggestion
   * Optional prop that is intended for any scenario where something is being generated by AI to reinforce AI transparency,
   * accountability, and explainability at the UI level.
```

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx [hunk 6]
```diff
@@ -664,6 +677,12 @@ TearsheetShell.propTypes = {
     })
   ),
 
+  /**
+   * Optional prop that is intended for any scenerio where something is being generated by AI to reinforce AI transparency,
+   * accountibility, and explainability at the UI level.
```
Comment (elycheea) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx:682]:
```suggestion
   * Optional prop that is intended for any scenario where something is being generated by AI to reinforce AI transparency,
   * accountability, and explainability at the UI level.
```

**Review by elycheea — COMMENTED** [review]
Just a few typos. “Scenario” and “accountability” were spelled slightly wrong but I think I caught them all. :)

**Comment by sangeethababu9223** [comment]
Thanks @elycheea!
The changes are updated now.

**Review by elycheea — APPROVED** [review]
Thank you! 🎉

**Review by elycheea — APPROVED** [review]

**Review by amal-k-joy — APPROVED** [review]
LGTM

**Review by anamikaanu96 — APPROVED** [review]
LGTM

**Pull request merged**